src/constraints.jl

```diff
@@ -187,26 +187,43 @@ Matrix given in compressed sparse row (CSR) format.
 `CSRMatrix` is analgous to the structure in Julia's `SparseMatrixCSC` but with
 the rows and columns flipped. It contains three vectors:
  - `row_pointers` is a vector pointing to the start of each row in
-    `columns` and `coefficients`;
+    `columns` and `coefficients` this should end with a `length(coefficients)+1`;
  - `columns` is a vector of column numbers; and
  - `coefficients` is a vector of corresponding nonzero values.
 
-The length of `row_pointers` is the number of rows in the matrix.
+The length of `row_pointers` is the number of rows in the matrix + 1.
 
 This struct is not a subtype of `AbstractSparseMatrix` as it is intended to be a
-collection of the three vectors as they are required by solvers such as Gurobi.
+collection of the three vectors as they are required by some solvers.
+Solvers such as Gurobi may need to trim the last index off of row_pointers.
 It is not intended to be used for general computation.
 """
 struct CSRMatrix{T}
+    # This whole definition should really be replaced by and Adjoint{SparseMatrixCSC}
```

Comment:
What are the arguments against doing this now?

Reply:
Nothing I can think of. I would like to hear from the broader team . 

From an implementation point of view what do you think would be better. A type Alias or encapsulation?